### DISCUSSION TITLE
Are Time.ticks global or per-VM?

### DISCUSSION THREAD

**Discussion by tve** [Moddable-OpenSource/moddable#1166] [Q&A] [answered]
Might be good to specify whether Time.ticks are global or per-VM. I'm writing a logger as a worker and I'm logging the ticks at each entry and now I'm wondering whether they're the same in the worker as in the main VM or whether they're perhaps relative to the start of the VM.
I believe on the esp32 a FreeRTOS function is called, so it would be global, but is that a property of the API or just of that specific implementation?

**Comment by phoddie** [comment, marked as answer]
All VMs use the same source of time. I'll update the docs to note:

> On devices that supports multiple concurrent JavaScript virtual machines (for example. using Wokers), the clock used to determine the value of the `ticks` property is the same across all virtual machines. This allows `tick` values created in one machine to be compared with values from another.   

Good?

**Reply by tve** [reply]
thanks!!

**Reply by phoddie** [reply]
I've committed changes to our local repo that will be available publicly soon. Docs updated and a test added.

On ESP32 the following tests all pass, confirming that `delta` behaves correctly around both wrap points.

```js
assert.sameValue(Time.delta(500, 600), 100);
assert.sameValue(Time.delta(500, 500), 0);
assert.sameValue(Time.delta(-1, 1), 2);
assert.sameValue(Time.delta(0, 0), 0);
assert.sameValue(Time.delta(-1, 0), 1);
assert.sameValue(Time.delta(0xFFFFFFFF, 1), 2);
assert.sameValue(Time.delta(0xFFFFFFFF, 0), 1);
assert.sameValue(Time.delta(0x7FFFFFFF, 0x80000001), 2);
assert.sameValue(Time.delta(0x7FFFFFFE, 0x80000000), 2);
assert.sameValue(Time.delta(+2147483647, -2147483647), 2);
assert.sameValue(Time.delta(+2147483646, -2147483648), 2);
```
